modules/dbnd-airflow/src/dbnd_airflow/airflow_extensions/request_factory.py

```diff
@@ -1,38 +1,71 @@
-from airflow.contrib.kubernetes.kubernetes_request_factory.pod_request_factory import (
-    SimplePodRequestFactory as AirflowSimplePodRequestFactory,
-)
+import logging
+import pprint
+
+from distutils.version import LooseVersion
+
+import airflow
+import yaml
+
 from airflow.contrib.kubernetes.secret import Secret
 
 
-class DbndPodRequestFactory(AirflowSimplePodRequestFactory):
+logger = logging.getLogger(__name__)
+
+
+class DbndPodRequestFactory(object):
+    def __init__(self, kubernetes_engine_config, pod_yaml):
+        self.kubernetes_engine_config = kubernetes_engine_config
+        self.yaml = yaml.safe_load(pod_yaml) or {}
+
     def create(self, pod):
-        req = super(DbndPodRequestFactory, self).create(pod=pod)
-        self.extact_extended_resources(pod, req)
+        original_pod = pod
+
+        if LooseVersion(airflow.version.version) > LooseVersion("1.10.10"):
+            pod = pod.to_v1_kubernetes_pod()
+
+        req = self.get_pod_serializer()(pod)
+
+        self.extract_node_affinity(original_pod, req)
+        self.extract_volume_secrets(original_pod, req)
+        self.extract_extended_resources(req)
+        self.extract_restart_policy(req)
+        logging.info("Created pod request: %s" % pprint.pprint(req))
```

Comment:
```suggestion
        logging.info("Created pod request: %s", pprint.pprint(req))
```
This allows log formatter format the arguments.

Reply:
Btw. does pprint return string?

Reply:
uh actually this is my log for debugging purposes, I'll remove it